internal/ls/completions.go

```diff
@@ -1186,10 +1192,10 @@ func (l *LanguageService) createCompletionItem(
 
 		// If is boolean like or undefined, don't return a snippet, we want to return just the completion.
 		if jsxAttributeCompletionStyleIs(preferences.JsxAttributeCompletionStyle, JsxAttributeCompletionStyleAuto) &&
-			t.Flags()&checker.TypeFlagsBooleanLike == 0 &&
-			!(t.Flags()&checker.TypeFlagsUnion != 0 && core.Some(t.Types(), func(t *checker.Type) bool { return t.Flags()&checker.TypeFlagsBooleanLike != 0 })) {
-			if t.Flags()&checker.TypeFlagsStringLike != 0 ||
-				t.Flags()&checker.TypeFlagsUnion != 0 &&
+			!t.IsBooleanLike() &&
+			!(t.IsUnion() && core.Some(t.Types(), func(t *checker.Type) bool { return t.IsBooleanLike() })) {
```

Comment:
```suggestion
			!(t.IsUnion() && core.Some(t.Types(), (*checker.Type).IsBooleanLike)) {
```

If you want to be clever.